pkg/config/nodetreemodel/config.go

```diff
@@ -190,6 +206,48 @@ func (c *ntmConfig) GetKnownKeysLowercased() map[string]interface{} {
 	return ret
 }
 
+// SetReady is called when Setup is complete, and the config is ready to be used
+func (c *ntmConfig) SetReady() {
+	defaultSourceRoot, err := NewNode(c.defaultBuilder, model.SourceDefault)
```

Comment:
How would NewNode knows the difference between inner and leaf Node ? Seems like setting any map as a default would build inner nodes instead of a leaf.

ex: `SetDefault("a,b", map[string]string{"c", "abc"}` would create a leaf node at `a.b.c="abc"`.

Reply:
You're right, this is already happening with `kubernetes_node_annotations_as_tags`. I'm going to work on fixing that.

Reply:
Going to save this for a future PR, as it's a bigger change than I think belongs in this one.